### PR TITLE
feat(stdlib/universe): added exponential moving average

### PR DESCRIPTION
**Pull request by roshie548**
### Done checklist
- [x] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

**Comment by roshie548** [comment]
`doInt`, `doUint`, and `doFloat` functions all share the same code. Any way to combine them into one? The problem I keep running into is I'm not sure how to generalize `*array.Int64`, `*array.Uint64`, etc.

**Comment by aanthony1243** [comment]
flux binary was somehow added in error again

File: stdlib/universe/exponential_moving_average.go [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+	mult := 2.0 / (float64(t.n) + 1)
+	count := 0.0
+	i := 0
+	var pValue float64
+	if vs.IsValid(i) {
+		pValue = float64(vs.Value(i))
+		count++
+	}
+	i++
+	for ; int64(i) < t.n; i++ {
```
[the remark this answers is not in the document]
Comment (roshie548) [reply stdlib/universe/exponential_moving_average.go]: This is outdated I believe

File: stdlib/testing/testdata/exponential_moving_average.flux [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+,result,table,_time,_value,_field,_measurement,device,fstype,host,path
+,,0,2018-05-22T00:00:20Z,30,used_percent,disk,disk1s1,apfs,host.local,/
+,,0,2018-05-22T00:00:30Z,35,used_percent,disk,disk1s1,apfs,host.local,/
+,,0,2018-05-22T00:00:40Z,37.5,used_percent,disk,disk1s1,apfs,host.local,/
+,,0,2018-05-22T00:00:50Z,38.75,used_percent,disk,disk1s1,apfs,host.local,/
+"
+
+exponential_moving_average = (table=<-) =>
+    (table
+        |> exponentialMovingAverage(n:3))
```
Comment (aanthony1243) [inline stdlib/testing/testdata/exponential_moving_average.flux]: add a range to this, as in  https://github.com/influxdata/flux/pull/1474/files

File: stdlib/universe/exponential_moving_average.go [hunk 3]
```diff
@@ -0,0 +1,576 @@
...
+			if err := b.AppendFloat(bj, t.pValue[bj]); err != nil {
+				return err
+			}
+		} else {
+			if err := b.AppendNil(bj); err != nil {
+				return err
+			}
+		}
+	}
+
```
Comment (roshie548) [inline stdlib/universe/exponential_moving_average.go]: Just realized...do I need to increment `t.i[bj]` and `j` here?

File: stdlib/universe/exponential_moving_average.go [hunk 4]
```diff
@@ -0,0 +1,576 @@
...
+				ema = (cValue * mult) + (t.pValue[bj] * (1.0 - mult))
+			}
+			if err := b.AppendFloat(bj, ema); err != nil {
+				return err
+			}
+			t.pValue[bj] = ema
+		}
+		t.i[bj]++
+	}
+	if int64(t.i[bj]) < t.n-1 {
```
Comment (roshie548) [inline stdlib/universe/exponential_moving_average.go]: Is this wrong? If the period is longer than the number of items in the chunk, this evaluates to true

**Pull request merged**